tools/calibration-96tof1/cal_eeprom/cal_map.py

```diff
@@ -0,0 +1,373 @@
+                     )  # write the nested key (Parameter key)
+                param_size, param_value = nested_value
+                eeprom_write_bytearray = eeprom_write_bytearray + \
+                    (struct.pack('<f', param_size))  # write the size of Param
+                for i in range(int(param_size/4)):
+                    eeprom_write_bytearray = eeprom_write_bytearray + \
+                        (struct.pack(
+                            '<f', param_value[i]))  # write the value of Param
+        eeprom_map_size = [self.get_map_size()]
+
+        eeprom_write_list = []
+        size = eeprom_write_bytearray.__len__()
+        for index in range(0, size):
+            eeprom_write_list.append(eeprom_write_bytearray[index])
+        #print("EEPROM WRITE List\n", eeprom_write_list)
+        #input("Press Enter to continue...")
+
+        size_list = []
+        size_byte = bytes()
+        size_byte = struct.pack('<f', size)
+        for index in range(0, 4):
+            size_list.append(size_byte[index])
+        eeprom.write(int(0), np.array(size_list, dtype='uint8'), 4)
+        eeprom.write(int(4), np.array(eeprom_write_list,
+                                      dtype='uint8'), eeprom_write_list.__len__())
+
+    def read_eeprom_cal_map(self, eeprom):
+        #print("Reading EEPROM")
+        data_array = np.zeros(4, dtype='uint8')
+        eeprom.read(int(0), data_array, 4)
+        read_size = struct.unpack('<f', data_array)
+        #print("Read Size",read_size)
+
+        data_array = np.zeros(int(read_size[0]), dtype='uint8')
+        eeprom.read(int(4), data_array, int(read_size[0]))
+        r_b = data_array.tobytes()
+
+        j = 0
+        while j < r_b.__len__():
+            key = r_b[j:j+4]
+            j = j+4
+            if not key:
+                break
+            key = struct.unpack('<f', key)
+            key = int(key[0])
+            #print("Primary Key", key)
+            sub_packet_size = struct.unpack('<f', r_b[j:j+4])
+            j = j+4
+            sub_packet_size = int(sub_packet_size[0])
+            #print("Sub Size",sub_packet_size)
+            sub_packet_map = {}
+            i = 0
+            while i < (sub_packet_size/4):  # 4:size of float
+                sub_packet_value = struct.unpack('<f', r_b[j:j+4])
+                j = j+4
+                sub_packet_value = int(sub_packet_value[0])
+                i = i + 1
+                parameter_key = sub_packet_value
+                #print("Param Key", parameter_key)
+
+                sub_packet_value = struct.unpack('<f', r_b[j:j+4])
+                j = j+4
+                sub_packet_value = int(sub_packet_value[0])
+                i = i + 1
+                parameter_size = sub_packet_value
+                #print("Param Size", parameter_size)
+
+                number_of_elements = int(parameter_size/4)  # 4:size of float
+                #print("Number of elements", number_of_elements)
```

Comment:
Are all these prints required to be here?

Reply:
not sure. I turned them into debug logs as they might be useful